R/aorsf_data.R

```diff
@@ -0,0 +1,225 @@
+# nocov start
+
+make_rand_forest_aorsf <- function(){
+
+
+  parsnip::set_model_engine("rand_forest", "classification", "aorsf")
+  parsnip::set_model_engine("rand_forest", "regression", "aorsf")
+  parsnip::set_dependency("rand_forest", "aorsf", "aorsf", mode = "classification")
+  parsnip::set_dependency("rand_forest", "aorsf", "aorsf", mode = "regression")
+
+  parsnip::set_model_arg(
+    model = "rand_forest",
+    eng = "aorsf",
+    parsnip = "mtry",
+    original = "mtry",
+    func = list(pkg = "dials", fun = "mtry"),
+    has_submodel = FALSE
+  )
+
+  parsnip::set_model_arg(
+    model = "rand_forest",
+    eng = "aorsf",
+    parsnip = "trees",
+    original = "n_tree",
+    func = list(pkg = "dials", fun = "trees"),
+    has_submodel = FALSE
+  )
+
+  parsnip::set_model_arg(
+    model = "rand_forest",
+    eng = "aorsf",
+    parsnip = "min_n",
+    original = "leaf_min_obs",
+    func = list(pkg = "dials", fun = "min_n"),
+    has_submodel = FALSE
+  )
+
+  parsnip::set_model_arg(
+    model = "rand_forest",
+    eng = "aorsf",
+    parsnip = "mtry",
+    original = "mtry",
+    func = list(pkg = "dials", fun = "mtry"),
+    has_submodel = FALSE
+  )
+
+  parsnip::set_fit(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "classification",
+    value = list(
+      interface = "formula",
+      protect = c("formula", "data", "weights"),
+      func = c(pkg = "aorsf", fun = "orsf"),
+      defaults =
+        list(
+          n_thread = 1,
+          verbose_progress = FALSE
+        )
+    )
+  )
+
+  parsnip::set_encoding(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "classification",
+    options = list(
+      predictor_indicators = "none",
+      compute_intercept = FALSE,
+      remove_intercept = FALSE,
+      allow_sparse_x = FALSE
+    )
+  )
+
+  parsnip::set_fit(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "regression",
+    value = list(
+      interface = "formula",
+      protect = c("formula", "data", "weights"),
```

Comment:
Protecting `weights` in `set_fit()` is tidymodels' way of declaring that an engine supports case weights. Looks like `orsf()` does indeed support case weights—could you add a test along the lines of [this](https://github.com/tidymodels/bonsai/blob/a91e80a384710206e07d91d6d90e11b344251b5a/tests/testthat/test-lightgbm.R#L766-L827) (or the [censored analogue](https://github.com/tidymodels/censored/blob/c03d80f1f49deea5c34206a834d7ed471bea4542/tests/testthat/test-rand_forest-aorsf.R#L270)) that confirms case weights have the same effect in tidymodels and the original orsf engine?

Reply:
Yes, will do!